scipy/stats/_rvs_sampling.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+import warnings
+from scipy._lib._util import check_random_state
+
+
+def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
+    """
+    Generate random samples from a probability density function using the
+    ratio-of-uniforms method.
+
+    Parameters
+    ----------
+    pdf : callable
+        A function with signature `pdf(x)` that is the probability
+        density function of the distribution.
+    vmin : float
+        The lower bound of the bounding rectangle in the x-direction.
+    vmax : float
+        The upper bound of the bounding rectangle in the x-direction.
+    umax : float
+        The upper bound of the bounding rectangle in the y-direction.
+    size : int or tuple of ints, optional
+        Defining number of random variates (default is 1).
+    c : float, optional.
+        Shift parameter of ratio-of-uniforms method, see Notes. Default is 0.
+    random_state : None or int or np.random.RandomState instance, optional
+        If already a RandomState instance, use it.
+        If seed is an int, return a new RandomState instance seeded with seed.
+        If None, use np.random.RandomState. Default is None.
+
+    Returns
+    -------
+    rvs : ndarray
+        The random variates distributed according to the probability
+        distribution defined by the pdf.
+
+    Notes
+    -----
+    Given a univariate probability density function (pdf) f and a constant c,
+    define the set ``A = {(u, v) : 0 < u <= sqrt(f(v/u + c))}``.
+    If ``(U, V)`` is a random vector uniformly distributed over ``A``,
+    then ``V/U + c`` follows a distribution with pdf ``f``.
```

Comment:
Be consistent in the markup used for f; currently the text has f, \`f\` and \`\`f\`\`.    I think \`f\` is fine (and use \`c\` for c, too).  The numpy docstring standard says to use single backticks for parameters, and f is not actually a parameter, but it is really just another name for `pdf`, so it might as well be treated the same.  Might as well use single backticks for the isolated occurrences of A and R, too, and just use the double backticks for expressions.  Text in single backticks gets rendered in italics, while double backticks are rendered as monospace text (i.e. code-like), so this isn't perfect, but I think that's a limitation of the reStructuredText that we're stuck with. (Well, it would be possible use :math: markup for all the mathematical variables and expressions, but personally, I view unrendered docstrings more often than I view the web versions, so I am biased against the heavy use of LaTeX markup.)

It might also work fine to remove *all* the backticks around the *mathematical* variables and expressions, and only use those for actual function parameter names and Python code, but I don't know if everyone would agree that that is acceptable.  (It definitely doesn't work if a variable is called `a`, but that is not the case here.) For example,

        define the set A = {(u, v) : 0 < u <= sqrt(f(v/u + c))}.

looks OK to me.

I'd write more, but it might end up as a mini-rant about the limitations of Sphinx and reStructuredText, so I'll stop. :)  The only change that is *really* needed here is to be consistent.